app/services/sso/openid_connect.rb

```diff
@@ -38,7 +38,15 @@ def bearer_token_set?
 
     def valid_issuer?
       payload = jwt_token.decoded_payload
-      payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
+      if payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
+        true
+      else
+        Foreman::Logging.logger('app').error "Failed to decode JWT: Invalid Issuer."
```

Comment:
I wonder if you want to log which issuer was received. Can make debugging easier.

Reply:
There is only one `oidc_issuer` provided in the settings, giving them the correct issuer might not be appropriate in terms of security. As in, they should know which issuer they are using. Does that make sense? let me know your thoughts.

Reply:
I meant the one that was received. So `payload['iss']` rather than the setting.

Reply:
I don't think this is a good idea, if we give away the `payload['iss']` away, users can easily change the value in the settings to the appropriate one. In my opinion, letting the user know that the issuer value is invalid is a good amount of hint.

Reply:
If users can change settings, isn't your installation already compromised? You can also log it at the debug level. That means a default installation won't show it but an admin has a way to figure out why it's not working. I can tell you that those things can be very frustrating.

As a more practical matter: I think the error should also be more specific. There's multiple things using JWT and technically speaking it has correctly decoded the JWT. It's just not valid. How about something like

```suggestion
        Foreman::Logging.logger('app').error "Invalid OIDC issuer received in JWT"
```

Reply:
makes sense!